sse2neon.h

```diff
@@ -6070,6 +6070,69 @@ FORCE_INLINE __m128 _mm_round_ps(__m128 a, int rounding)
+// element in the lower element of dst, and copy the upper 3 packed elements
+// from a to the upper elements of dst. Rounding is done according to the
+// rounding[3:0] parameter, which can be one of:
+//     (_MM_FROUND_TO_NEAREST_INT |_MM_FROUND_NO_EXC) // round to nearest, and
+//     suppress exceptions
+//     (_MM_FROUND_TO_NEG_INF |_MM_FROUND_NO_EXC)     // round down, and
+//     suppress exceptions
+//     (_MM_FROUND_TO_POS_INF |_MM_FROUND_NO_EXC)     // round up, and suppress
+//     exceptions
+//     (_MM_FROUND_TO_ZERO |_MM_FROUND_NO_EXC)        // truncate, and suppress
+//     exceptions _MM_FROUND_CUR_DIRECTION // use MXCSR.RC; see
+//     _MM_SET_ROUNDING_MODE
+// https://software.intel.com/sites/landingpage/IntrinsicsGuide/#text=_mm_round_ss
+FORCE_INLINE __m128 _mm_round_ss(__m128 a, __m128 b, int rounding)
+{
+#if defined(__aarch64__)
+    switch (rounding) {
+    case (_MM_FROUND_TO_NEAREST_INT | _MM_FROUND_NO_EXC):
+        return vreinterpretq_m128_f32(_mm_move_ss(
+            vreinterpretq_f32_m128(a), vrndnq_f32(vreinterpretq_f32_m128(b))));
+    case (_MM_FROUND_TO_NEG_INF | _MM_FROUND_NO_EXC):
+        return vreinterpretq_m128_f32(_mm_move_ss(
+            vreinterpretq_f32_m128(a), vrndmq_f32(vreinterpretq_f32_m128(b))));
+    case (_MM_FROUND_TO_POS_INF | _MM_FROUND_NO_EXC):
+        return vreinterpretq_m128_f32(_mm_move_ss(
+            vreinterpretq_f32_m128(a), vrndpq_f32(vreinterpretq_f32_m128(b))));
+    case (_MM_FROUND_TO_ZERO | _MM_FROUND_NO_EXC):
+        return vreinterpretq_m128_f32(_mm_move_ss(
+            vreinterpretq_f32_m128(a), vrndq_f32(vreinterpretq_f32_m128(b))));
+    default:  //_MM_FROUND_CUR_DIRECTION
+        return vreinterpretq_m128_f32(_mm_move_ss(
+            vreinterpretq_f32_m128(a), vrndiq_f32(vreinterpretq_f32_m128(b))));
+    }
+#else
+    float *_a = (float *) &a;
+    float *_b = (float *) &b;
+    __m128 zero, neg_inf, pos_inf;
+
+    switch (rounding) {
+    case (_MM_FROUND_TO_NEAREST_INT | _MM_FROUND_NO_EXC):
+        return _mm_move_ss(vreinterpretq_f32_m128(a),
+                           _mm_cvtepi32_ps(_mm_cvtps_epi32(b)));
+    case (_MM_FROUND_TO_NEG_INF | _MM_FROUND_NO_EXC):
+        return (__m128){floorf(_b[0]), _a[1], _a[2], _a[3]};
+    case (_MM_FROUND_TO_POS_INF | _MM_FROUND_NO_EXC):
+        return (__m128){ceilf(_b[0]), _a[1], _a[2], _a[3]};
+    case (_MM_FROUND_TO_ZERO | _MM_FROUND_NO_EXC):
+        zero = _mm_set_ps(0.0f, 0.0f, 0.0f, 0.0f);
+        neg_inf = _mm_set_ps(floorf(_b[0]), floorf(_b[1]), floorf(_b[2]),
+                             floorf(_b[3]));
+        pos_inf =
+            _mm_set_ps(ceilf(_b[0]), ceilf(_b[1]), ceilf(_b[2]), ceilf(_b[3]));
+        return _mm_move_ss(
+            vreinterpretq_f32_m128(a),
+            _mm_blendv_ps(pos_inf, neg_inf, _mm_cmple_ps(b, zero)));
+    default:  //_MM_FROUND_CUR_DIRECTION
+        return (__m128){roundf(_b[0]), _a[1], _a[2], _a[3]};
```

Comment:
The original implementation from `_mm_round_ps()` is wrong.
It would be affected by the `_MM_SET_ROUNDING_MODE()` and the corresponding implementation should be different.